### PR TITLE
Add conditional CI execution

### LINKED ISSUE #609: Add conditional check specific to a file extension.
**Is your feature request related to a problem? Please describe.**
The CI is really slow.

**Describe the solution you'd like**
Add conditional check (path check).

**Describe alternatives you've considered**
...

**Additional context**
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**
* code scanning / Clang => edited c files
* code scanning / CodeQL-cpp => edited c files
* code scanning / CodeQL-python => edited python files
* code scanning / CodeQL-javascript => edited javascript files
* Mixed linter and checks / cmd_descs_yaml_check => edited yaml files
* Mixed linter and checks / clang-format => edited c files
* Mixed linter and checks / prettier => edited javascript files
* CI / * => edited c files
* TinyCC build / ubuntu-tcc-test => edited c files

PS: I include side effect files (\*\*/meson.build, shlr/rizin-shell-parser/\*\*, subprojects/\*\*, .github/workflows/code-analysis.yml) for all build related task.

**Test plan**
...

**Closing issues**
close https://github.com/rizinorg/rizin/issues/609

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,6 +3,14 @@ name: CI
 
 on:
   push:
+    paths:
```
Comment (ret2libc) [inline .github/workflows/ci.yml:6]: all the meson.build files?

Comment (ghost) [reply .github/workflows/ci.yml:6]: oh, I forgot.

**Comment by ghost** [comment]
I need help for everything related to sourcehut.
And don't merge, I need to test if it is possible to merge if a required test is skipped

**Comment by ghost** [comment]
> I need help for everything related to sourcehut.

@XVilka 

> And don't merge, I need to test if it is possible to merge if a required test is skipped

I test and it works.

Travis isn't configurable to skip build.

**Comment by XVilka** [comment]
I think SourceHut doesn't support this yet, just keep it as is. They are still in the alpha and will improve in the future.

**Comment by ghost** [comment]
> I think SourceHut doesn't support this yet, just keep it as is. They are still in the alpha and will improve in the future.

Okay, I believe everything is ready. Feel free to notify me if I miss a ci workfow.

**Review by ret2libc — APPROVED** [review]
If you tested it and it works even when jobs are mark as required by github, then i'm fine to merge this!!

**Comment by ghost** [comment]
> If you tested it and it works even when jobs are mark as required by github, then i'm fine to merge this!!

I will test again, we never know.

**Comment by ghost** [comment]
2021-02-13-201025_664x308_scrot

It is working

**Pull request merged**